web-ng/src/app/components/layer-dialog/layer-dialog.component.ts

```diff
@@ -146,9 +146,17 @@ export class LayerDialogComponent implements OnDestroy {
     }
     this.form = this.layerService.getForm(this.layer);
     if (this.form) {
-      this.fields =
+      const fields =
         this.form?.fields.toList().sortBy(field => field.index) ||
         List<Field>();
+      fields.forEach((field, ind) => {
```

Comment:
Instead of rebuilding Fields with sorted multiple choice options, perhaps it's simpler to sort them in the constructor of MultipleChoice? This way the options in a give Field will always be in order. Wdyt?